packer/builder/azure/arm/step_capture_image.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+type StepCaptureImage struct {
+	client  *AzureClient
+	capture func(resourceGroupName string, computeName string, parameters *compute.VirtualMachineCaptureParameters) error
+	say     func(message string)
+	error   func(e error)
+}
+
+func NewStepCaptureImage(client *AzureClient, ui packer.Ui) *StepCaptureImage {
+	var step = &StepCaptureImage{
+		client: client,
+		say:    func(message string) { ui.Say(message) },
+		error:  func(e error) { ui.Error(e.Error()) },
+	}
+
+	step.capture = step.captureImage
+	return step
+}
+
+func (s *StepCaptureImage) captureImage(resourceGroupName string, computeName string, parameters *compute.VirtualMachineCaptureParameters) error {
+	generalizeResponse, err := s.client.Generalize(resourceGroupName, computeName)
```

Comment:
Correct.  There's a distinct build step to make it generalized - I reuse the code from existing builder.